File: app/src/main/java/com/platdmit/simplecloudmanager/data/api/ApiDomainRepo.java
package com.platdmit.simplecloudmanager.data.api;

import com.platdmit.simplecloudmanager.data.api.models.ApiDomain;
import com.platdmit.simplecloudmanager.data.api.models.ApiDomainRecord;

import java.util.List;

import io.reactivex.rxjava3.core.Single;

public interface ApiDomainRepo {
    Single<List<ApiDomain>> getDomains();
    Single<ApiDomain> getDomain(long domainId);
    Single<List<ApiDomainRecord>> getDomainRecords(long domainId);
    Single<ApiDomainRecord> getDomainRecord(int domainId, int recordId);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/data/api/ApiDomainRepo.kt
package com.platdmit.simplecloudmanager.data.api

import com.platdmit.simplecloudmanager.data.api.models.ApiDomain
import com.platdmit.simplecloudmanager.data.api.models.ApiDomainRecord
import io.reactivex.rxjava3.core.Single

interface ApiDomainRepo {
    fun getDomains(): Single<List<ApiDomain>>
    fun getDomain(domainId: Long): Single<ApiDomain>
    fun getDomainRecords(domainId: Long): Single<List<ApiDomainRecord>>
    fun getDomainRecord(domainId: Int, recordId: Int): Single<ApiDomainRecord>
}